models/issue_tracked_time_test.go

```diff
@@ -0,0 +1,96 @@
+package models
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddTime(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+	//3661 = 1h 1min 1s
+	trackedTime, err := AddTime(3, 1, 3661)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(3), trackedTime.UserID)
+	assert.Equal(t, int64(1), trackedTime.IssueID)
+	assert.Equal(t, int64(3661), trackedTime.Time)
+
+	tt := AssertExistsAndLoadBean(t, &TrackedTime{UserID: 3, IssueID: 1}).(*TrackedTime)
+	assert.Equal(t, tt.Time, int64(3661))
+
+	comment := AssertExistsAndLoadBean(t, &Comment{Type: CommentTypeAddTimeManual, PosterID: 3, IssueID: 1}).(*Comment)
+	assert.Equal(t, comment.Content, "1h 1min 1s")
+}
+
+func TestGetTrackedTimes(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+
+	// by Issue
+	times, err := GetTrackedTimes(FindTrackedTimesOptions{IssueID: 1})
+	assert.Len(t, times, 1)
+	assert.Equal(t, times[0].Time, int64(400))
+	assert.NoError(t, err)
+
+	times, err = GetTrackedTimes(FindTrackedTimesOptions{IssueID: -1})
+	assert.Len(t, times, 0)
+	assert.NoError(t, err)
+
+	// by User
+	times, err = GetTrackedTimes(FindTrackedTimesOptions{UserID: 1})
+	assert.Len(t, times, 1)
+	assert.Equal(t, times[0].Time, int64(400))
+	assert.NoError(t, err)
```

Comment:
Could we please check `err` before checking `times`? Checking for errors before checking other return values makes tests easier to debug.

Using this test as an example, suppose that `err` is not `nil`. Then `times` will most likely be an empty slice, and line 41 will result in an index-out-of-bounds panic. This panic will occur before the `assert.NoError(t, err)` check, so in this case we will have no way of knowing whether `GetTrackedTimes(..)` returned an empty slice because of an error, or for some other reason.

Likewise in other places.